pkg/ruler/ruler.go

```diff
@@ -43,6 +43,12 @@ var (
 		Name:      "blocked_workers",
 		Help:      "How many workers are waiting on an item to be ready.",
 	})
+	evalLatency = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: "cortex",
+		Name:      "group_evaluation_latency_seconds",
+		Help:      "How far behind the target time each rule group executed.",
+		Buckets:   []float64{.1, .25, .5, 1, 2.5, 5, 10, 25},
```

Comment:
Really looking forward to this metric! That said, is 25 seconds for the histogram max long enough? You seem to hint in https://github.com/weaveworks/cortex/issues/680 that the ruler can potentially be hours behind, and I have seen it be several minutes behind.

Reply:
Histograms have a cap on detailed quartiles but also give you a mean which is uncapped. 

Happy to review after we get some real data to look at.

Reply:
👍